### ISSUE TITLE
more varied style and layout schemes

### ISSUE THREAD

**Issue by brainchild0** [gingko/client#176]
Liking the underlying concept of Gingko, I experimented with some composition.  In my case, I found the visual appearance to be an obstacle that I would  not normally experience in other environments, despite the obvious advantages of Gingko.

For other individuals  the experience will obviously vary.

For me, two primary   obstacles manifested:

1.  The particular stylization of blue background and broad card borders is not an optimal  design for helping me focus on  the text.
1.  The composition  always appears in a tree layout, which by design greatly facilitates composition and organization, but makes it difficult to review the flow as it would appear to the reader.

The first issue might be addressed by  a skinnable interface.  But separate from producing such an interface and inviting users to contribute skins, I would be really happy if an option were available to change all the  background  colors to white and to display the card borders as a minimal pale gray outline.

The second would require more effort to resolve.  Ideally, the effect would be to introduce a display mode such that the document appears as continuous flowing text, as in a browser window, but can be edited as in  a traditional text buffer while still retaining  the hierarchical card structure.

For some, "full screen mode" might be very useful, but for me the ideas above would bring far more benefit toward the objective of being able to write as the thoughts flow without being influenced by environmental distractions.

I suppose these suggestions are more abstract than the ones typically presented, but  could they  be discussed?

**Comment by AdrianoFerrari** [comment]
Hi there,

Thanks for the input. I find myself wanting this as well, in particular the "display as continuous text" mode (if you have more to add about the colors/themes, please do so in issue #80).

Can you give me more details on your particular use case?
1. Would you want to be able to edit a *column* in this new display mode?
Or would it be the entire document *tree*?
2. And at what stage do you find yourself needing/wanting this?

**Comment by SquidLord** [comment]
Actually, while we're discussing this – the ability to display the current working document as a continuous text which is composed only of the rightmost nodes in the tree of every branch off the route would be extremely helpful.

Primarily for the reason that I often work in a very snowflake-like method and the intermediate nodes before the rightmost are simply elements which get broken down into the eventual intentional text itself. They have no meaning along the way as document content. And if an intermediate node currently doesn't have any branches to the right, it's fine to put that as a block into the document.

Being able to edit backwards, that is editing the presented "final document" and having nodes update could be a little complicated, especially at node boundaries, but with some sort of subtle presentation of what the breaks between nodes are in the continuous text frame, it might work out all right.

**Comment by AdrianoFerrari** [comment]
You're right, displaying the rightmost or "leaf" nodes of the tree in this continuous form is also an important one to consider.

In fact, it might be good to combine this idea, fullscreen mode (#10), and the "export live preview" idea (#51), into one feature, calling it "linear view mode" or something.

This "linear view mode" would be low on distractions, closer to the final output, and configurable to display any of the export options available (full tree, single column, rightmost only). From there the user could make changes in that view, or export them to continue working in a different enviroment.

**Comment by brainchild0** [comment]
> Can you give me more details on your particular use case?
> 
> 1. Would you want to be able to edit a _column_ in this new display mode?
>    Or would it be the entire document _tree_?
> 2. And at what stage do you find yourself needing/wanting this?

I'm not sure I understand the questions, exactly.  By "display a column", do you mean display the results of a tree traversal that excludes depth lower than some particular value?  I suppose certain filter rules such as this one or the one @SquidLord suggests *may* be a useful enhancement at some time, but i wasn't thinking about such a thing when opening this discussion.

The basic forms of the concept could be represented in practical terms by any of the following, in increasing order of difficulty to implement:


1.  A preview mode, showing an noneditable representation of the current paragraph flow.
1.  Above, but editable such that changes in the text buffer are reflected in the corresponding cards.
1.  Above, with card boundaries visible, either for all cards or for the one located a the current cursor position.
1.  Above, with a tree representation on the side, possibly foldable, allowing visualization of the hierarchy while still presenting the flowing paragraph form.
1.  Above, with ability for adding, removing, moving, and copying cards.

For  me, this effect becomes useful as soon as the text of a work or chapter reaches a few paragraphs, or a few hundred words.  In an academic paper with a comparatively distinctive top-down structure,   the early stages of development might rely more on filling existing cards and have reduced need to a feature such as this one.

Filtering is a separate and complex subject that probably needs to be considered elsewhere.  I myself can think of some case where I would use it, though not necessarily the same case as have been mentioned just now.

**Comment by brainchild0** [comment]
Since stylization discussion is captured in #80, I am changing title to more accurately reflect the reduced scope of this issue topic.

**Comment by brainchild0** [comment]
> You're right, displaying the rightmost or "leaf" nodes of the tree in this continuous form is also an important one to consider.

Forgive me, but I am missing the rationale for this idea. Why is this view mode useful? Would not the effect be to display a redacted document, that is, one with only some cards displayed, and others hidden?

> 
> In fact, it might be good to combine this "continuous flowing text", fullscreen mode (#10), and the "export live preview" idea (#51), into one feature, calling it "linear flow mode" or something.
> 
> This "linear flow mode" would be low on distractions, closer to the final output, and configurable to display any of the export options available (full tree, single column, rightmost only). From there the user could make changes in that view, or export them to continue working in a different environment.
> 

Maybe, but these functions may be separate concerns that ought to be accessed in separate features. It seems as though the thinking is that certain combinations of parameters will be particularly useful to the user. It may be beneficial offer a handful of display modes that simultaneously set a group of parameters to coordinating values, but it may equally be a mistake to prevent fine-grained configuration of distinctive parameters. 


> The key design challenge would be keeping enough of the link to the underlying card-based document to be able to make changes in this mode that feel natural and correspond to the expected changes in the tree, without becoming too complicated or confusing.

Yes, this challenge is certainly the central concern in any case.

**Comment by brainchild0** [comment]
Although not new information to many, I learned, after finally investigating the features and usage of Scrivener, that it offers a display option called Scrivenings Mode, which, essentially the same as the features proposed in item (2) of posting #5 above, could provide a point of reference. 

The view mode further provides an outline representation in the sidebar, but in my trials, I find that the integration between the two views, tree and text, though presented side-by-side, is much more limited than might be optimal.
